linkerd/app/outbound/src/http/concrete.rs

```diff
@@ -353,14 +353,19 @@ impl<T> svc::Param<tls::ConditionalClientTls> for Endpoint<T> {
         self.metadata
             .identity()
             .cloned()
-            .map(move |server_id| {
+            .map(move |client_tls| {
                 let alpn = if use_transport_header {
                     use linkerd_app_core::transport_header::PROTOCOL;
                     Some(tls::client::AlpnProtocols(vec![PROTOCOL.into()]))
                 } else {
                     None
                 };
-                tls::ConditionalClientTls::Some(tls::ClientTls::new(server_id, alpn))
+
+                tls::ConditionalClientTls::Some(tls::ClientTls::new(
+                    client_tls.server_id,
+                    client_tls.server_name,
+                    alpn,
+                ))
```

Comment:
alternately -- we could add a `ClientTls::with_alpn(self, alpn: Option<AlpnProtocols>) -> ClientTls` if you wanted to avoid the mut, so this would end up 

```rust
           .map(move |mut client_tls| {
                tls::ConditionalClientTls::Some(client_tls.with_alpn(if use_transport_header {
                    use linkerd_app_core::transport_header::PROTOCOL;
                    Some(tls::client::AlpnProtocols(vec![PROTOCOL.into()]))
                } else {
                    None
                }))